Author: aszlig <aszlig@nix.build>
Date:   Thu Jun 28 15:26:53 2018 +0200

    migration/task-history: Reuse Mongoose conn info
    
    Instead of using the hardcoded default values for Monk, let's actually
    use the info from Mongoose to connect to the database using Monk.
    
    We only really care about the host, because we're connecting using Unix
    sockets so we don't need any auth information.
    
    Signed-off-by: aszlig <aszlig@nix.build>
    Filename: fix-connection-info-for-task-history-migration.patch

diff --git a/migrations/tasks/habits-one-history-entry-per-day-challenges.js b/migrations/tasks/habits-one-history-entry-per-day-challenges.js
index dcb05f48b..12e55ecb7 100644
--- a/migrations/tasks/habits-one-history-entry-per-day-challenges.js
+++ b/migrations/tasks/habits-one-history-entry-per-day-challenges.js
@@ -6,10 +6,11 @@
  * Iterates over all habits and condense multiple history entries for the same day into a single entry
  */
 
+const mongoose = require('mongoose');
 const monk = require('monk');
 const _ = require('lodash');
 const moment = require('moment');
-const connectionString = 'mongodb://localhost:27017/habitrpg?auto_reconnect=true'; // FOR TEST DATABASE
+const connectionString = `mongodb://${mongoose.connection.host}`;
 const dbTasks = monk(connectionString).get('tasks', { castIds: false });
 
 function processChallengeHabits (lastId) {
diff --git a/migrations/tasks/habits-one-history-entry-per-day-users.js b/migrations/tasks/habits-one-history-entry-per-day-users.js
index 8eb52e8f9..4add6702f 100644
--- a/migrations/tasks/habits-one-history-entry-per-day-users.js
+++ b/migrations/tasks/habits-one-history-entry-per-day-users.js
@@ -6,10 +6,11 @@ const authorUuid = 'ed4c688c-6652-4a92-9d03-a5a79844174a'; // ... own data is do
  * Iterates over all habits and condense multiple history entries for the same day into a single entry
  */
 
+const mongoose = require('mongoose');
 const monk = require('monk');
 const _ = require('lodash');
 const moment = require('moment');
-const connectionString = 'mongodb://localhost:27017/habitrpg?auto_reconnect=true'; // FOR TEST DATABASE
+const connectionString = `mongodb://${mongoose.connection.host}`;
 const dbTasks = monk(connectionString).get('tasks', { castIds: false });
 const dbUsers = monk(connectionString).get('users', { castIds: false });
 
